spec/controllers/api/current/analytics/user_analytics_controller_spec.rb

```diff
@@ -1,45 +1,85 @@
 require 'rails_helper'
 
 RSpec.describe Api::Current::Analytics::UserAnalyticsController, type: :controller do
-  render_views
-
   let!(:request_user) { create(:user) }
   let!(:request_facility) { create(:facility, facility_group: request_user.facility.facility_group) }
 
-  let!(:patients) { create_list :patient, 10, registration_facility: request_facility, registration_user: request_user }
-
   before :each do
     request.env['HTTP_X_USER_ID'] = request_user.id
     request.env['HTTP_X_FACILITY_ID'] = request_facility.id
     request.env['HTTP_AUTHORIZATION'] = "Bearer #{request_user.access_token}"
   end
 
-  describe 'GET: send data from server to device;' do
-    it 'gets data for 4 weeks as a hashmap' do
-      get :show, format: :json
+  describe '#show' do
+    context 'json_api' do
+      describe 'facility has no patients registered' do
+        it 'gets data for as a hashmap' do
+          get :show, format: :json
 
-      response_body = JSON(response.body)
-      expect(response_body).to be_instance_of(Hash)
-    end
+          response_body = JSON(response.body)
+          expect(response_body).to be_nil
+        end
+      end
 
-    it 'gets html when requested' do
-      get :show, format: :html
+      describe 'facility has patients registered' do
+        let!(:patients) { create_list(:patient, 2, registration_facility: request_facility) }
 
-      expect(response.status).to eq(200)
-      expect(response.body).to match(/analytics-container/)
-      expect(response.body).to match(/featured-graph/)
+        it 'returns the statistics for the facility as json' do
+          get :show, format: :json
+
+          response_body = JSON(response.body)
+          expect(response_body.keys.map(&:to_sym))
+            .to include(:first_of_current_month,
+                        :total_patients_count,
+                        :unique_patients_per_month,
+                        :patients_enrolled_per_month)
+        end
+      end
     end
 
-    it 'has the facility name' do
-      get :show, format: :html
+    context 'html api' do
+      render_views
 
-      expect(response.body).to match(Regexp.new(request_facility.name))
-    end
+      describe 'facility has no patients registered' do
+        it 'returns an empty graph' do
```

Comment:
These tests can maybe just be if the correct templates / partials are being rendered instead of a piece of some content from each view?

Reply:
Added https://github.com/rails/rails-controller-testing for this. 
Updated the spec.